docs/governance/rfc/002-go-cors-authn.md

```diff
@@ -0,0 +1,56 @@
+# 002: CORS and authn for Go
+
+This RFC proposes adding two ancillary Go projects to the Connect Github
+organization:
+
+* `connectrpc.com/cors`, which eases server-side CORS configuration, and
+* `connectrpc.com/authn`, which provides flexible server-side authentication
+  middleware.
+
+## CORS
+
+[Cross-origin resource sharing][cors], commonly called CORS, allows servers to
+indicate whether web browsers should allow scripts to load resources from other
+domains. This is a common concern for single-page web applications, where the
+API and app are often hosted on different origins &mdash; for example,
+`app.acme.com` may need to call APIs hosted on `api.acme.com`. As part of its
+security model, CORS requires servers to allowlist individual request and
+response headers. 
+
+For backend-to-backend communication, the Connect runtime abstracts away
+protocol-specific headers. But to correctly configure header allowlists for
+CORS, server authors must first understand quite a bit of the gRPC, gRPC-Web,
+and Connect protocols. The [resulting configuration][explicit-cors] is verbose
+and difficult to evolve with the underlying protocols.
+
+To make development of browser-facing Connect APIs easier in Go, we propose
+creating a small Go package of CORS helpers. This package will help users
+configure existing CORS packages (for example, [`github.com/rs/cors`][rs-cors]) without
+needing to explicitly list all the HTTP headers used by each RPC protocol.
+
+## Authentication
+
+HTTP servers use a variety of authentication schemes: mutual TLS, cookies,
+and various types of bearer tokens are particularly common. Often, the
+authentication logic also requires some knowledge of the service schema &mdash;
+at least the name of the service and method, but sometimes more detailed
+information too.
+
+In Go, authentication checks are best implemented as `net/http` middleware.
```

Comment:
Would agree.

Do we intend to provide some built-in implementations for things like Basic auth, or mTLS, etc? Or is the goal to just provide the abstraction to implement their own?

Reply:
Just an abstraction to implement their own. We can always add more functionality later on :)